### ISSUE TITLE
Composite string not detected

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/python-jsonpreprocessor#102]
This is the content of JSON file:

```
"par1"   : "par1",
"par2"   : "par2",
"par3"   : ["A", "B"],
"par4"   : ["ara"],
"ara"    : "ara",
"param"  : "param",
"param2" : ${par3},
//
"val2" : ${par3}[0]${p${par4}[0]m2}[1]
```

`val2` consists of a composite string. The first part is represented by `${par3}[0]`, the second part is represented by `${p${par4}[0]m2}[1]`.

Because of the string is a composite one, it is expected that the entire expression in encapsulated in quotes. The quotes are missing, therefore I expect an error message, telling about these missing quotes. But I do not expect returned values.

Instead of this the JsonPreprocessor returns values. The value of `val2` is: '`ara`'.

When I add the missing quotes:

`"val2" : "${par3}[0]${p${par4}[0]m2}[1]"`

the JsonPreprocessor also returns values, but now this is expected. But the value of `val2` now is: '`A`'.

Both returned values are wrong. In second case (with quotes around expression) the expected value for `val2` is '`AB`'.

**Comment by namsonx** [comment]
Hello Holger, Hello Thomas,

This issue was fixed in stabi branch ([namsonx/task/stabi_branch](https://github.com/test-fullautomation/python-jsonpreprocessor/tree/namsonx/task/stabi_branch))

Thank you,
Son

**Comment by HolQue** [comment]
Retest successful

**Comment by test-fullautomation** [comment]
solved and already merged with #147
